### PR TITLE
Fix suppression of the `using` statement

### LINKED ISSUE #57: `@suppress_err using` doesn't work since v0.2.3
In Suppressor v0.2.3  https://github.com/JuliaIO/Suppressor.jl/pull/53 has introduced `with_logger`, and this seems to make `@suppress_err using xxx` not working anymore, e.g.,

```julia
julia> using Suppressor

julia> @suppress_err using YAML
ERROR: syntax: "using" expression not at top level
Stacktrace:
 [1] top-level scope
   @ REPL[2]:1
```

This works fine in Suppressor v0.2.2.

Is this supposed to be a valid usage? If so, we might need to find a way to patch this.

cc: @Drvi @quinnj

### PR DESCRIPTION
**Pull request by Drvi**
fixes #57

### CODE REVIEW

File: src/Suppressor.jl [hunk 1]
```diff
@@ -37,11 +37,14 @@ macro suppress(block)
             _logger = current_logger()
         end
 
+        # Spelling out `with_logger(f, ...)` as a try/finally so we can do `@suppress using SpammyPkg`
```
Comment (NHDaly) [inline src/Suppressor.jl]: Is this because otherwise the expression wouldn't be top-level?

Comment (NHDaly) [reply src/Suppressor.jl]: If so, can you clarify that in the comment?
I.e.: when i read your comment, i still don't know _why_ we need to spell it out to make `using` work.

File: src/Suppressor.jl [hunk 2]
```diff
@@ -37,11 +37,14 @@ macro suppress(block)
...
+        t = current_task()
+        old_logstate = t.logstate
         try
-            Logging.with_logger(_logger) do
-                $(esc(block))
-            end
+            t.logstate = Base.CoreLogging.LogState(_logger)
+            $(esc(block))
         finally
+            t.logstate = old_logstate
```
Comment (NHDaly) [inline src/Suppressor.jl:48]: I think a nice alternative could be to write a small `@_with_logger(l)`  macro that expands to the same thing as what you have here, which would reduce your code duplication.

What do you think about that?

Comment (NHDaly) [reply src/Suppressor.jl:48]: Though i know there are some weirdnesses with calling one macro from another one.. :/ But i think it's not too bad? I don't remember the specifics though.

**Review by NHDaly — APPROVED** [review]
LGTM, with or without my suggestion for a macro to reduce duplication.

File: src/Suppressor.jl (hunk 2 above)
Comment (Drvi) [reply src/Suppressor.jl:48]: Thanks! I can try but my macro-fu is not that strong.

File: src/Suppressor.jl (hunk 1 above)
Comment (Drvi) [reply src/Suppressor.jl]: Yes, exactly, I'll add a comment

File: src/Suppressor.jl (hunk 2 above)
Comment (Drvi) [reply src/Suppressor.jl:48]: I just couldn't figure the macro in a macro out

**Comment by johnnychen94** [comment]
@quinnj could you make a release for this? Thanks in advance!

**Comment by quinnj** [comment]
Kicked off a new release.

**Pull request merged**